### PR TITLE
Feat/1207192574958023 peaq staking mechanism

### PR DESCRIPTION
**Pull request by sfffaaa**

### CODE REVIEW

**Comment by sfffaaa** [comment]
Pending because found some performance issues. Need to redesign the way to calculation/distribute tokens

**Comment by sfffaaa** [comment]
We'll continue to solve the performance issue by delay distribution

File: pallets/parachain-staking/src/lib.rs [hunk 1]
```diff
@@ -2713,8 +2821,25 @@ pub mod pallet {
...
+		///		1.3. Calcuate the reward:
+		///			collator reward = collator reward ratio * pot balance
+		///			delegator reward = delegator reward ratio * pot balance
+		///		1.4. Transfer the reward to the collator and delegator.
+		/// 2. we need to clean up the state of the pallet.
+		fn end_session(end_index: SessionIndex) {
+			log::debug!("new_session: {:?}", end_index);
+			Self::peaq_reward_mechanism_impl();
+
+			CollatorBlock::<T>::iter().for_each(|(k, _)| {
```
Comment (DocteurPing) [inline pallets/parachain-staking/src/lib.rs]: Can we use `drain` method here to remove all the elements ?

Comment (sfffaaa) [reply pallets/parachain-staking/src/lib.rs]: Good suggestion! Already modify it

**Review by DocteurPing — APPROVED** [review]

**Pull request merged**